slices/slice_func.go

```diff
@@ -191,6 +191,18 @@ func Map[S ~[]T, T any, G any](items S, f func(el T) G) []G {
 	return result
 }
 
+// Map applies F to all elements in slice of T and returns slice results as pointers, and will filter out the non-nil ones.
+func MapFilter[S ~[]T, T any, G any](items S, f func(el T) *G) []G {
```

Comment:
Maybe, using a separate bool argument would be more clean?

1. Returning the pointer can get messy when `G` itself is a pointer. Then you'd need to return `**int`.
2. Returning the bool would make it looks similar to how Filter works.

```suggestion
func MapFilter[S ~[]T, T any, G any](items S, f func(el T) (G, bool)) []G {
```